lib/Doctrine/ODM/MongoDB/UnitOfWork.php

```diff
@@ -1297,7 +1298,6 @@ private function cascadePostUpdateAndPostPersist(ClassMetadata $class, $document
                     } else {
                         if ( ! empty($entryClass->lifecycleCallbacks[Events::postUpdate])) {
                             $entryClass->invokeLifecycleCallbacks(Events::postUpdate, $entry);
-                            $this->recomputeSingleDocumentChangeSet($entryClass, $entry);
```

Comment:
@jwage: I'm afraid I need to pick your brain on this. Left in place, this line causes collections being updated through atomic strategies to get re-scheduled for updates, simply for having a postUpdate lifecycle callback (even if it does not modify the data).

I traced this line back to one of your old commits in: https://github.com/doctrine/mongodb-odm/commit/b7455dcec8ebf83dea2339bdbc7981db3c622c75#diff-ae2c9f4b58c612dce08a0f60ff683aecR893

Do you have any idea why you were recomputing changesets for postUpdate lifecycle callbacks, but not general postUpdate event listeners (immediately below) or postPersist events (both lifecycle and general, which happen a few lines up)?